legend-engine-xt-json-model/src/main/java/org/finos/legend/engine/external/format/json/JsonExternalFormatExtension.java

```diff
@@ -62,48 +46,24 @@ public Root_meta_external_format_json_metamodel_JsonSchema compileSchema(Externa
         return new JsonSchemaCompiler(context).compile();
     }
 
```

Comment:
Seems like jsonSchema supports both model and schema generation, could we add that to return true and the properties. 
````suggestion

    @Override
    public boolean supportsModelGeneration()
    {
        return true;
    }

    @Override
    public RichIterable<? extends Root_meta_pure_generation_metamodel_GenerationParameter> getModelGenerationProperties(PureModel pureModel)
    {
        return core_external_format_json_binding_jsonSchemaToPure.Root_meta_external_format_json_binding_toPure_describeConfiguration__GenerationParameter_MANY_(pureModel.getExecutionSupport());
    }

````

Was testing this in studio and its missing this for jsonSchema to generate models

Reply:
Interface is changed, this is no longer needed to added explicitly, this should come from implemented interfaces

```
public class JsonExternalFormatExtension implements ExternalFormatSchemaGenerationExtension<Root_meta_external_format_json_metamodel_JsonSchema, ModelToJsonSchemaConfiguration>, ExternalFormatModelGenerationExtension<Root_meta_external_format_json_metamodel_JsonSchema, JsonSchemaToModelConfiguration>
```

This is what is used in description api as well, not sure why this does not come up in studio?

Reply:
Maybe I am missing something but when i load your PR for jsonSchema I get the below 
```
{
  "contentTypes": ["application/json"],
  "modelGenerationProperties": [],
  "name": "JSON",
  "schemaGenerationProperties": [],
  "supportsModelGeneration": false,
  "supportsSchemaGeneration": false
}
```

Studio uses the `supportsModelGeneration` to decide whether to show the model generation editor. 
XSD returns that as true and the properties

Reply:
On launching server with this PR changes and hitting this api
```
http://127.0.0.1:6060/api/pure/v1/external/format/availableFormats
```

I get following response
```
[
{
"contentTypes": [
"application/json"
],
"modelGenerationProperties": [
{
"defaultValue": "JSON",
"description": "",
"name": "format",
"required": true,
"type": "STRING"
},
{
"defaultValue": "null",
"description": "The id of the schema within the SchemaSet that will drive the generation",
"name": "sourceSchemaId",
"required": false,
"type": "STRING"
},
{
"defaultValue": "target::package",
"description": "The package name into which the generated model classes will be created",
"name": "targetPackage",
"required": true,
"type": "STRING"
}
],
"name": "JSON",
"schemaGenerationProperties": [
{
"defaultValue": "JSON",
"description": "",
"name": "format",
"required": true,
"type": "STRING"
},
{
"defaultValue": "target::package::GeneratedSchemaSet",
"description": "Full name of the SchemaSet to be generated",
"name": "targetSchemaSet",
"required": true,
"type": "STRING"
}
],
"supportsModelGeneration": true,
"supportsSchemaGeneration": true
},
{
"contentTypes": [
"application/x.flatdata"
],
"modelGenerationProperties": [
{
"defaultValue": "FlatData",
"description": "",
"name": "format",
"required": true,
"type": "STRING"
},
{
"defaultValue": "null",
"description": "The id of the schema within the SchemaSet that will drive the generation",
"name": "sourceSchemaId",
"required": false,
"type": "STRING"
},
{
"defaultValue": "target::package",
"description": "The package name into which the generated model classes will be created",
"name": "targetPackage",
"required": true,
"type": "STRING"
},
{
"defaultValue": "true",
"description": "Make names in generated model into conventional names",
"name": "purifyNames",
"required": true,
"type": "BOOLEAN"
},
{
"defaultValue": "null",
"description": "The name of the class to represent the whole schema if one is to be generated",
"name": "schemaClassName",
"required": false,
"type": "STRING"
}
],
"name": "FlatData",
"schemaGenerationProperties": [
{
"defaultValue": "FlatData",
"description": "",
"name": "format",
"required": true,
"type": "STRING"
},
{
"defaultValue": "target::package::GeneratedSchemaSet",
"description": "Full name of the SchemaSet to be generated",
"name": "targetSchemaSet",
"required": true,
"type": "STRING"
}
],
"supportsModelGeneration": true,
"supportsSchemaGeneration": true
},
{
"contentTypes": [
"application/protobuf",
"application/vnd.google.protobuf"
],
"modelGenerationProperties": [ ],
"name": "Protobuf",
"schemaGenerationProperties": [
{
"defaultValue": "Protobuf",
"description": "",
"name": "format",
"required": true,
"type": "STRING"
},
{
"defaultValue": "target::package::GeneratedSchemaSet",
"description": "Full name of the SchemaSet to be generated",
"name": "targetSchemaSet",
"required": true,
"type": "STRING"
},
{
"defaultValue": "null",
"description": "",
"name": "javaPackage",
"required": false,
"type": "STRING"
},
{
"defaultValue": "null",
"description": "The class name (and hence the file name) for the wrapper Java class you want to generate. If no explicit java_outer_classname is specified in the .proto file, the class name will be constructed by converting the .proto file name to camel-case (so foo_bar.proto becomes FooBar.java). If the java_multiple_files option is disabled, then all other classes/enums/etc. generated for the .proto file will be generated within this outer wrapper Java class as nested classes/enums/etc.",
"name": "javaOuterClassname",
"required": false,
"type": "STRING"
},
{
"defaultValue": "null",
"description": "If false, only a single .java file will be generated for this .proto file, and all the Java classes/enums/etc. generated for the top-level messages, services, and enumerations will be nested inside of an outer class (see java_outer_classname). If true, separate .java files will be generated for each of the Java classes/enums/etc. generated for the top-level messages, services, and enumerations, and the wrapper Java class generated for this .proto file won\\'t contain any nested classes/enums/etc. This is a Boolean option which defaults to false",
"name": "javaMultipleFiles",
"required": false,
"type": "BOOLEAN"
},
{
"defaultValue": "null",
"description": "Affects the C++ and Java code generators (and possibly third-party generators) changing the generated code to optimize for these characteristics.",
"items": {
"enums": [
"meta::external::format::protobuf::transformation::fromPure::OptimizeMode.SPEED",
"meta::external::format::protobuf::transformation::fromPure::OptimizeMode.CODE_SIZE",
"meta::external::format::protobuf::transformation::fromPure::OptimizeMode.LITE_RUNTIME"
],
"types": [
"STRING"
]
},
"name": "optimizeFor",
"required": false,
"type": "ENUM"
},
{
"defaultValue": "null",
"description": "Freeform options that are not part of the specification but can be passed through to other tooling",
"items": {
"enums": [ ],
"types": [
"STRING",
"STRING"
]
},
"name": "customOptions",
"required": false,
"type": "MAP"
}
],
"supportsModelGeneration": false,
"supportsSchemaGeneration": true
},
{
"contentTypes": [
"application/xml"
],
"modelGenerationProperties": [
{
"defaultValue": "XSD",
"description": "",
"name": "format",
"required": true,
"type": "STRING"
},
{
"defaultValue": "null",
"description": "The id of the schema within the SchemaSet that will drive the generation",
"name": "sourceSchemaId",
"required": false,
"type": "STRING"
},
{
"defaultValue": "target::package",
"description": "The package name into which the generated model classes will be created",
"name": "targetPackage",
"required": true,
"type": "STRING"
},
{
"defaultValue": "true",
"description": "Should classes that represent only a collection be eliminated from the structure? For example, if true, <things><thing/><thing/></things> becomes Thing[*] rather than a class Things with a single property of type Thing[*].",
"name": "inlineCollectionClasses",
"required": true,
"type": "BOOLEAN"
},
{
"defaultValue": "false",
"description": "Should classes derived from types included in the XSD but not reachable from the document root still be generated? These are usually structures included into the XSD but not used in the specific document(s) described.",
"name": "includeUnreachableClasses",
"required": true,
"type": "BOOLEAN"
}
],
"name": "XSD",
"schemaGenerationProperties": [ ],
"supportsModelGeneration": true,
"supportsSchemaGeneration": false
},
{
"contentTypes": [
"application/json"
],
"modelGenerationProperties": [
{
"defaultValue": "GraphQL_Introspection",
"description": "",
"name": "format",
"required": true,
"type": "STRING"
},
{
"defaultValue": "null",
"description": "The id of the schema within the SchemaSet that will drive the generation",
"name": "sourceSchemaId",
"required": false,
"type": "STRING"
},
{
"defaultValue": "target::package",
"description": "The package name into which the generated model classes will be created",
"name": "targetPackage",
"required": true,
"type": "STRING"
}
],
"name": "GraphQL_Introspection",
"schemaGenerationProperties": [ ],
"supportsModelGeneration": true,
"supportsSchemaGeneration": false
}
]
```

Reply:
I think i was outdated a bit. Yeah I see the code now. Thanks